dataRetriever/storageResolvers/headerResolver.go

```diff
@@ -79,7 +79,7 @@ func (hdrRes *headerResolver) RequestDataFromHash(hash []byte, _ uint32) error {
 	metaEpoch := hdrRes.epochHandler.MetaEpoch()
 	hdrRes.mutEpochHandler.RUnlock()
 
-	hdrRes.manualEpochStartNotifier.NewEpoch(metaEpoch + 1)
+	hdrRes.manualEpochStartNotifier.NewEpoch(metaEpoch + 2)
```

Comment:
will this work always ?

Reply:
yes

Reply:
it opens in advance 2 persisters rather than 1. Works on the 0->1 and x->x+1 edge cases as tested with a fetched import-db archive